File: domain/src/main/java/com/study/platform/domain/post/model/PostTitle.java
package com.study.platform.domain.post.model;

import com.study.platform.global.exception.CustomException;
import com.study.platform.global.exception.ErrorCode;
import jakarta.persistence.Embeddable;

@Embeddable
public record PostTitle(String value) {

    public PostTitle {
        if (value == null || value.isBlank()) {
            throw new CustomException(ErrorCode.INVALID_INPUT);
        }
        if (value.length() > 100) {
            throw new CustomException(ErrorCode.INVALID_INPUT);
        }
    }
}

File: domain/src/main/kotlin/com/study/platform/domain/post/model/PostTitle.kt
package com.study.platform.domain.post.model

import com.study.platform.global.exception.CustomException
import com.study.platform.global.exception.ErrorCode
import jakarta.persistence.Embeddable

@Embeddable
class PostTitle(val value: String) {
    init {
        if (value.isBlank()) throw CustomException(ErrorCode.INVALID_INPUT)
        if (value.length > 100) throw CustomException(ErrorCode.INVALID_INPUT)
    }
}
